.dev_scripts/test_init_backbone.py

```diff
@@ -0,0 +1,280 @@
+    config_path = _get_config_directory()
+    check_cfg_names = []
+
+    # `base`, `legacy_1.x` and `common` ignored by default.
+    ignore_cfg_names = ['_base_', 'legacy_1.x', 'common']
+    # 'ld' need load teacher model, if want to check 'ld',
+    # please check teacher_config path first.
+    ignore_cfg_names += ['ld']
+    # `selfsup_pretrain` need convert model, if want to check this model,
+    # need to convert the model first.
+    ignore_cfg_names += ['selfsup_pretrain']
+
+    # the `init_cfg` in 'centripetalnet', 'cornernet', 'cityscapes' and
+    # 'scratch' is None.
+    # Please confirm `bockbone.init_cfg` is None first.
+    ignore_cfg_names += [
+        'centripetalnet', 'cornernet', 'cityscapes', 'scratch'
+    ]
+
+    for config_file_name in os.listdir(config_path):
+        if config_file_name in ignore_cfg_names:
+            continue
+        config_file = join(config_path, config_file_name)
+        if os.path.isdir(config_file):
+            for config_sub_file in os.listdir(config_file):
+                if config_sub_file.endswith('py'):
+                    name = join(config_file, config_sub_file)
+                    check_cfg_names.append(name)
+    return check_cfg_names
+
+
+def _check_backbone(config, print_cfg=True):
+    """Check out backbone whether successfully load pretrained model, by using
+    `backbone.init_cfg`.
+
+    First, using `mmcv._load_checkpoint` to load the checkpoint without
+        loading models.
+    Then, using `build_detector` to build models, and using
+        `model.init_weights()` to initialize the parameters.
+    Finally, assert weights and bias of each layer loaded from pretrained
+        checkpoint are equal to the weights and bias of original checkpoint.
+        For the convenience of comparison, we sum up weights and bias of
+        each loaded layer separately.
+
+    Args:
+        config (str): Config file path.
+        print_cfg (bool): Whether print logger and return the result.
+
+    Returns:
+        results (str or None): If backbone successfully load pretrained
+            checkpoint, return None; else, return config file path.
+    """
+    if print_cfg:
+        print('-' * 15 + 'loading ', config)
+    cfg = Config.fromfile(config)
+    init_cfg = None
+    try:
+        init_cfg = cfg.model.backbone.init_cfg
+        init_flag = True
+    except AttributeError:
+        init_flag = False
+    if init_cfg is None or init_cfg.get('type') != 'Pretrained':
+        init_flag = False
+    if init_flag:
+        checkpoint = _load_checkpoint(init_cfg.checkpoint)
+        model = build_detector(
+            cfg.model,
+            train_cfg=cfg.get('train_cfg'),
+            test_cfg=cfg.get('test_cfg'))
+        model.init_weights()
+
+        layers = []
+        names = []
+        checkpoint_names = []
+        layer_indexes = []
+        darknet_indexs = []
+        for name in model.backbone.state_dict():
```

Comment:
We can simplify the logic by directly comparinging the `state_dict` and `checkpoitn`